floodviz/static/js/hydrograph.js

```diff
@@ -1,36 +1,11 @@
-document.addEventListener("DOMContentLoaded", function (event) {
-	"use strict";
+"use strict";
 
-	var data_path = FV.hydrograph_data_path;
+FV.hydromodule = ( function (options) {
 
-	// Set the dimensions of the canvas / graph
+	var self = {}
 	var margin = {top: 30, right: 20, bottom: 30, left: 50};
-	var width = FV.chart_dimensions.width - margin.left - margin.right;
-	var height = FV.chart_dimensions.height - margin.top - margin.bottom;
-
-	// Set the ranges
-	var x = d3.scaleTime().range([0, width]);
-	var y = d3.scaleLog().range([height, 0]);
-
-	// Define the voronoi
-	var voronoi = d3.voronoi()
-		.x(function (d) {
-			return x(d.time_mili);
-		})
-		.y(function (d) {
-			return y(d.value);
-		})
-		.extent([[-margin.left, -margin.top], [width + margin.right, height + margin.bottom]])
-
-	// Define the line
-	var line = d3.line()
-		.x(function (d) {
-			return x(d.time_mili);
-		})
-		.y(function (d) {
-			return y(d.value);
-		});
-
+	var height = options.height - margin.top - margin.bottom;
+	var width = options.width - margin.left - margin.right;
 	// Adds the svg canvas
 	var svg = d3.select("#hydrograph")
```

Comment:
It would be more reusable and testable if you pass in the id of the div where the hydrograph is created.